core/src/services/hdfs/backend.rs

```diff
@@ -302,7 +303,7 @@ impl Access for HdfsBackend {
     async fn write(&self, path: &str, op: OpWrite) -> Result<(RpWrite, Self::Writer)> {
         let target_path = build_rooted_abs_path(&self.root, path);
```

Comment:
Similarly, we can move parts of this implementation to core as `HdfsCore::hdfs_open_path_for_writing`. Once we have done that, the `write` function will become:

- open the file for writing
- return results (`HdfsWriter` is what OpenDAL returns for users to write data from)